docs/developer-docs/docs/operate-a-node/buenavista-testnet/upgrade/v0.4.2.md

```diff
@@ -0,0 +1,110 @@
+﻿---
+sidebar_position: 2
+---
+
+# v0.4.2
+
+## Overview
+
+This guide provides steps for upgrading to the [Warden v0.4.2](https://github.com/warden-protocol/wardenprotocol/releases/tag/v0.4.2) release, which makes changes to `keychainFees` to be non nullable.
+
+The network upgrade will take place on **September 2 2024**. The Warden team is going to take the snapshot, export the state, and revert it back if needed.
+
+If you have any outstanding questions, [join our Discord](https://discord.com/invite/warden).
+
+:::tip
+Stay tuned for the precise upgrade date and additional details to be clarified.
+:::
+
+:::warning
+The upgrade procedure carries heightened risks of double-signing and being slashed. To handle these risks and securely address any issues that may arise during the upgrade, [back up](#1-verify--back-up) your data prior to the upgrade. During the upgrade, follow our tips listed in [Security and troubleshooting](#security--troubleshooting).
+:::
+
+## 1. Verify & back up
+
+Before starting your validator and signing, do the following:
+
+- Verify your software version and the genesis file hash.
+- Back up your **private node keys**.
+- Take a **full data snapshot**.
+
+The method for taking a snapshot depends on your infrastructure, but generally, you can achieve this by backing up the **wardend** directory. If you're [using Cosmovisor](#option-1-use-cosmovisor) to upgrade, it'll back up your data by default.
+
+:::warning
+After stopping the `wardend` process, it's critically important to back up the `.wardend/data/priv_validator_state.json` file.
+:::
+
+**Note**: The **priv_validator_state.json** file is updated every block as your validator participates in consensus rounds. In case the upgrade fails, and you have to restart the previous chain, you'll need this file to avoid double signing. See [Handling node restarts](#handling-node-restarts).
+
+## 2. Run Connect sidecar
+
+Prior to the update, you should also run the **Skip:Connect** binary on the same machine as `wardend`. Make sure you enable it from the [configuration](/operate-a-node/operate-connect#3-configure-wardend).
+
+Follow this guide: [Operate Connect](/operate-a-node/operate-connect).
+
+## 3. Upgrade
+
+### Option 1: Use Cosmovisor
+
+To upgrade using [Cosmovisor](https://pkg.go.dev/cosmossdk.io/tools/cosmovisor), take these steps:
+
+1. Download the `wardend` v0.4.2 binary and put it to `$WARDEN_HOME/cosmovisor/upgrades/v042/bin`:
+
+   ```shell
+   mkdir -p  $WARDEN_HOME/cosmovisor/upgrades/v042/bin
+   cp $(which wardend) $WARDEN_HOME/cosmovisor/upgrades/v042/bin
+   ```
+
+   **Note**: Alternatively, enable auto-download in Cosmovisor, and it'll automatically download the binary.
+
+2. Start the node:
+
+   ```shell
+   cosmovisor run start --x-crisis-skip-assert-invariants --home $DAEMON_HOME
```

Comment:
```suggestion
   cosmovisor run start --home $DAEMON_HOME
```

Reply:
Why do we recommend doing this? :thinking:

Reply:
Yeah - agreed. We don't need to skip state check.